pymc3/sampling.py

```diff
@@ -55,6 +92,10 @@ def sample(draws, step, start=None, trace=None, chain=0, njobs=1, tune=None,
     -------
     MultiTrace object with access to sampling values
     """
+    model = modelcontext(model)
+    
+    step = assign_step_methods(model, step)
```

Comment:
One worry is that this will always assign step methods. I've sometimes wanted to sample only a few variables and leave the others fixed. This seems like that makes that hard. 

One fix would be to supply a 'do nothing' sampler. There's probably other things we could do too.